query-engine/connector-test-kit-rs/query-engine-tests/tests/new/native_upsert.rs

```diff
@@ -0,0 +1,242 @@
+use query_engine_tests::*;
+
+#[test_suite(capabilities(NativeUpsert))]
+mod native_upsert {
```

Comment:
Now that you're using `resolve_compound_field`, adding a test for a compound `@@id[]` would be great 👍